clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java

```diff
@@ -1330,12 +1346,21 @@ protected static String getValidateSchemaAndBuildDictionaryOutputFileName(String
         + VALIDATE_SCHEMA_AND_BUILD_DICTIONARY_MAPPER_OUTPUT_FILE_EXTENSION;
   }
 
-  private void getValidateSchemaAndBuildDictMapperOutput(JobConf conf, String jobId) throws Exception {
-    String outputDir = getValidateSchemaAndBuildDictionaryOutputDir(conf, pushJobSetting.storeName);
+  private void getValidateSchemaAndBuildDictMapperOutput(String jobId) throws Exception {
+    String outputDir = getValidateSchemaAndBuildDictionaryOutputDir(
+        pushJobSetting.useMapperToBuildDictOutputPath,
+        pushJobSetting.storeName);
     String outputAvroFile = getValidateSchemaAndBuildDictionaryOutputFileName(jobId);
-    validateSchemaAndBuildDictMapperOutputReader =
-        new ValidateSchemaAndBuildDictMapperOutputReader(outputDir, outputAvroFile);
-    inputFileDataSize = validateSchemaAndBuildDictMapperOutputReader.getInputFileDataSize() * INPUT_DATA_SIZE_FACTOR;
+    ValidateSchemaAndBuildDictMapperOutputReader outputReader = null;
+    try {
+      outputReader = new ValidateSchemaAndBuildDictMapperOutputReader(outputDir, outputAvroFile);
+      validateSchemaAndBuildDictMapperOutput = outputReader.getOutput();
+    } finally {
+      if (outputReader != null) {
+        outputReader.close();
+      }
+    }
```

Comment:
This can be reduced to:
```
try (ValidateSchemaAndBuildDictMapperOutputReader outputReader = new ValidateSchemaAndBuildDictMapperOutputReader(outputDir, outputAvroFile)) {
    validateSchemaAndBuildDictMapperOutput = outputReader.getOutput();
}
```

Reply:
Thanks, Modified it.